sympy/simplify/simplify.py

```diff
@@ -1101,7 +1101,11 @@ def tofunc(nu, z):
 
     def tominus(fro):
         def tofunc(nu, z):
-            return exp(I*pi*nu)*fro(nu, exp_polar(-I*pi)*z)
+            c, s = cos(S.Pi*nu), sin(S.Pi*nu)
+            if not isinstance(c, cos) and not isinstance(s, sin):
+                return (c + S.ImaginaryUnit)*fro(nu, exp_polar(-I*pi)*z)
```

Comment:
It seems that `s` is missing from this expression. Maybe that is the reason why the answer was different.